### DISCUSSION TITLE
List of Java Objects

### DISCUSSION THREAD

**Discussion by katiaxyz** [rodnansol/spring-configuration-property-documenter#81] [Q&A]
Any idea about how to document a list of Java objects?
Let's say we have:

```
@Data
@ConfigurationProperties(prefix = "my-prefix")
public class MyProperties {
  private List<Person> people;
}
```

the configuration would be:

```
my-prefix.people[0].name = "Xyz"
my-prefix.people[1].name = "Yxz"
my-prefix.people[0].age = 2
my-prefix.people[1].age = 3
```

if Person has fields name and age.

spring-boot-configuration-processor simply generates a spring-configuration-metadata.json where the field people is of type List<Person>

Would it be possible to generate sample configuration like above ? Or is there any suggestion?

**Comment by nandorholozsnyak** [comment]
Hello there,

I'll take a look on it but as we know, Java at runtime has no information about the generic's type because of type erasure, so this might cause the problem with the annotation processor.

Hand written json files can be written, they are named `additional-spring-configuration-metadata.json` in the META-INF folder, maybe you can take a look on it.

https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#appendix.configuration-metadata.annotation-processor.adding-additional-metadata

**Reply by katiaxyz** [reply]
Yes I have used the additional-spring-configuration-metadata.json to add "kind of definition" of the pojo's properties though it is a feasible approach only when the pojo has just a few fields.
